Uses HTTP POST for Solr calls in the catalog controller. Fixes #13

hectorcorrea opened the pull request:
@awead @cam156 Could you take a look at the `search_layout` method that I defined in catalog_controller.rb? 

Sufia 3.x used to provide a `search_layout` method but that seems to be gone from Sufia 5.x. Do you have any ideas on what is the new way of getting the search layout? Do we still show a different layout based on the search parameters?

mjgiarlo commented:
We used to need `search_layout` because both the homepage and the dashboard were both Blacklight controllers, I believe, but now the homepage has its own controller that is not Blacklight-based. You can see this method simplified here:

https://github.com/projecthydra/sufia/commit/8c5d97644b4fd40c5ede4d8ba32fde6e661f7e3d

And you can see the two-column layout added to `Sufia::Catalog` here:

https://github.com/projecthydra/sufia/commit/f145aa45a43c7087ba6aef97a36cdcb4d84d1ae0

And then the method is removed here:

https://github.com/projecthydra/sufia/commit/0c3477537a36ac9e81f7622096f063b3591a75a3

So since Sufia 4.0.0, you should no longer need this method.

hectorcorrea commented:
@mjgiarlo thanks for the feedback. I removed the `layout :search_layout` altogether from the controller. The test still passes and the search to Solr is now using HTTP POST.

The controller blows up when running the search inside the application but that's unrelated to the layout. I've added ticket #43 for it.
